[SAASINT-4161] New Integration for Workato: Initial manifest as first step in provisioning (#20296)
DataDog/integrations-core: workato/README.md

# Agent Check: Workato

## Overview

> [Workato][1] provides a powerful platform for enterprise automation and integration so you can confidently connect, integrate, and automate your applications, data, and processes end-to-end.

Reviewer: ```suggestion
[Workato][1] provides a platform for enterprise automation and integration so you can connect, integrate, and automate your applications, data, and processes end-to-end.
```
Removed marketing language.

> The Workato integration collects Jobs Logs (Recipe execution) and Status Metrics (Connection and Recipe state), sending them to Datadog for detailed analysis. The logs are parsed and enriched for efficient searching, while the metrics provide insights into operational status.

Reviewer 2: Are there vendor links to recipe execution? Example: https://docs.workato.com/en/recipes/running-recipes.html

> The integration includes a dashboard that shows Job execution status and duration, making it easier to monitor and understand trends and issues.

Reviewer 2: [nit] Should Job be capitalized

Author: Will use lower case for Workato elements like recipes and jobs.

## Setup

### Generate API credentials in Workato

1. Log in to [Workato][2] as an administrator. 
2. Navigate to **Workspace Admin > API Clients**.
3. Click **Create API Client**, and then enter the required information:
   4. Under **Name**, Enter a descriptive and identifiable client name.

>    5. Choose an appropriate **Client role** from the drop-down menu. Ensure the selected role has LIST and GET DETAILS permissions on all Workato Resources.

Reviewer 2: ```suggestion
   5. Choose an appropriate **Client role** from the drop-down menu. Ensure the selected role has `LIST` and `GET DETAILS` permissions on all Workato Resources.
```

>    5. Choose an appropriate **Client role** from the dropdown - ensure the selected role has LIST and GET DETAILS permissions on all Workato Resources.

Reviewer: ```suggestion
   5. Choose an appropriate **Client role** from the drop-down menu. Ensure the selected role has LIST and GET DETAILS permissions on all Workato Resources.
```

>    6. Choose between _Selected Projects_ or _All Projects_ from the **Project access** dropdown.

Reviewer: ```suggestion
   6. Choose between _Selected Projects_ or _All Projects_ from the **Project access** drop-down menu.
```

>    7. Leave the **Allowed IPs** text box blank/empty to allow access from any IP.

Reviewer: ```suggestion
   7. Leave the **Allowed IPs** text box blank to allow access from any IP.
```

> 8. Click **Create Client** and copy the API token from here for later use.       

Reviewer: ```suggestion
8. Click **Create Client**, and copy the API token for later use.       
```

### Connect your Workato Account to Datadog

1. Add your Access Token
    |Parameters|Description|
    |--------------------|--------------------|
    |Access Token|Access token for your HubSpot private app.|
2. Click the Save button to save your settings.

## Data Collected

### Logs 

The Workato integration collects and forwards Activity logs to Datadog.

### Metrics

> The HubSpot Content Hub integration collects and forwards Analytics metrics to Datadog.

Author: Thanks, fixed.

# Agent Check: Workato

## Overview

[Workato][1] provides a platform for enterprise automation and integration so you can connect, integrate,
and automate your applications, data, and processes end-to-end.

> The Workato integration collects [jobs](https://docs.workato.com/en/recipes/jobs.html#viewing-job-reports) logs
> (recipe execution) and status metrics (connection and recipe state), sending them to Datadog for detailed analysis.

Reviewer: ```suggestion
The Workato integration collects [jobs](https://docs.workato.com/en/recipes/jobs.html#viewing-job-reports), logs
(recipe execution), and status metrics (connection and recipe state), sending them to Datadog for detailed analysis.
```

The logs are parsed and enriched for efficient searching, while the metrics provide insights into operational status.

The integration includes a dashboard that shows job execution status and duration, making it easier to monitor and
understand trends and issues.

## Setup

### Generate API credentials in Workato

1. Log in to [Workato][2] as an administrator. 
2. Navigate to the **Workspace Admin** tab. > API Clients**.
3. (Create a Client Role with sufficient permission) Click **Client Roles** tab.

> 4. Client the **+ Add client role** button.

Reviewer: ```suggestion
4. Click **+ Add client role**.
```

   1. Check the following boxes to allow minimal read access:
   ```
      Projects
           Project Assets
               List Projects
               List Folders
           Connections
               List
           Recipes
               List
               Get job counts for recipes
               Get details
           Recipe Versions
               List
           Jobs
               List
               Get job
      Admin
            Environment Management
                Tags
                    List tags
   ```

>    2. Edit the role name and click **Save changes**.

Reviewer: ```suggestion
   2. Edit the role name, and click **Save changes**.
```

> 7. Client the **API Clients** tab.

Reviewer: ```suggestion
7. Select the **API Clients** tab.
```

8. Click **+ Add API Client**, and then enter the required information:

>    1. Under **Name**, Enter a descriptive and identifiable client name.

Reviewer: ```suggestion
   1. Under **Name**, enter a descriptive and identifiable client name.
```

>    2. Choose the newly created Client Role from the dropdown or ensure the selected role has `LIST` and `GET DETAILS` permissions on all Workato Resources.

Reviewer: ```suggestion
   2. Choose the newly created Client Role from the dropdown, or ensure the selected role has `LIST` and `GET DETAILS` permissions on all Workato Resources.
```
